wsbroadcastserver/clientconnection.go

```diff
@@ -76,49 +101,167 @@ func (cc *ClientConnection) Compression() bool {
 	return cc.compression
 }
 
+// Register sends the ClientConnection to be registered with the ClientManager.
+func (cc *ClientConnection) Register() {
+	cc.clientAction <- ClientConnectionAction{
+		cc:     cc,
+		create: true,
+	}
+}
+
+// Remove sends the ClientConnection to be removed from the ClientManager.
+func (cc *ClientConnection) Remove() {
+	cc.clientAction <- ClientConnectionAction{
+		cc:     cc,
+		create: false,
+	}
+}
+
+func (cc *ClientConnection) writeBacklog(ctx context.Context, segment backlog.BacklogSegment) error {
+	var prevSegment backlog.BacklogSegment
+	for !backlog.IsBacklogSegmentNil(segment) {
+		// must get the next segment before the messages to be sent are
+		// retrieved ensures another segment is not added in between calls.
+		prevSegment = segment
+		segment = segment.Next()
+
+		select {
+		case <-ctx.Done():
+			return errContextDone
+		default:
+		}
+
+		msgs := prevSegment.Messages()
+		if prevSegment.Contains(uint64(cc.requestedSeqNum)) {
+			requestedIdx := int(cc.requestedSeqNum) - int(prevSegment.Start())
+			msgs = msgs[requestedIdx:]
+		}
+		bm := &m.BroadcastMessage{
+			Version:  1, // TODO(clamb): I am unsure if it is correct to hard code the version here like this? It seems to be done in other places though
```

Comment:
We should probably add a constant for this in some common package

Reply:
I added a constant for this in `broadcaster/messages` lib, let me know if you like it to be somewhere else/named something else